src/passkey_child/passkey_child_assert.c

```diff
@@ -34,6 +34,59 @@
 
 #include "passkey_child.h"
 
+errno_t
+get_assert_user_id(TALLOC_CTX *mem_ctx, fido_assert_t *assert,
+                   unsigned char **_user_id)
+{
+    TALLOC_CTX *tmp_ctx = NULL;
+    size_t user_id_len;
+    const unsigned char *user_id;
+    char *b64_user_id;
+    errno_t ret;
+
+    tmp_ctx = talloc_new(NULL);
+    if (tmp_ctx == NULL) {
+        ERROR("talloc_new() failed\n");
+        return ENOMEM;
+    }
+
+    user_id_len = fido_assert_user_id_len(assert, 0);
+    if (user_id_len == 0) {
+        DEBUG(SSSDBG_TRACE_FUNC, "There isn't any user id\n");
+        ret = EOK;
+        goto done;
+    } else if (user_id_len != USER_ID_SIZE) {
```

Comment:
Hi,

I'm not sure if this should be treated as an error. If the device was registered with a different utility than `passkey_child`  the user id might have a different length. Or is the length defined in some specs?

bye,
Sumit

Reply:
Hi,

I'm just guessing, maybe there will be environments where it will be similar as with Smartcards today where the devices will be configured centrally (maybe even externally) and you are not allowed to add new keys.

bye,
Sumit

Reply:
In that case, is it okay if we print an informative message, but do not return an error?

Reply:
Debug message. Something like:
```
DEBUG(SSSDBG_TRACE_FUNC,
              "User id size %ld doesn't match expectation %d. This could lead to errors when getting the user id.\n",
              user_id_len, USER_ID_SIZE);
```